adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/HttpClientBuilder.java

```diff
@@ -287,16 +289,20 @@ public HttpClient build() {
                 cm = new SingleClientConnManager(registry);
             }
             BasicHttpParams params = new BasicHttpParams();
-            if (socketTimeout > -1)
-            {
+
+            if (proxyHost != null) {
+                params.setParameter(ConnRoutePNames.DEFAULT_PROXY, proxyHost);
```

Comment:
Just gave it a quick spin - turns out that I'd probably need to port the whole HTTP Client config to the new HttpClientBuilder API.

I think that is a bit out of scope for this PR. I'd rather go with the small change (using the deprectaed but working API) and port the HTTP config in a separate PR. wdyt?